src/Tasks/ResolveAssemblyReferences/Server/RarController.cs

```diff
@@ -0,0 +1,130 @@
+﻿using Microsoft.Build.Internal;
+using Microsoft.Build.Shared;
+using System;
+using System.IO;
+using System.IO.Pipes;
+using System.Security.AccessControl;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+
+using StreamJsonRpc;
+using Microsoft.Build.Tasks.ResolveAssemblyReferences;
+using System.Collections.Generic;
+using Nerdbank.Streams;
+using Microsoft.VisualStudio.Threading;
+using System.Diagnostics;
+using Microsoft.Build.Tasks.ResolveAssemblyReferences.Contract;
+using Microsoft.Build.Tasks.ResolveAssemblyReferences.Services;
+
+#nullable enable
+namespace Microsoft.Build.Tasks.ResolveAssemblyReferences.Server
+{
+    public sealed class RarController
+    {
+        private const int PipeBufferSize = 131072;
+        
+        private readonly string _pipeName;
+
+        private readonly IResolveAssemblyReferenceTaskHandler _rarTaskHandler;
+
+        private NamedPipeServerStream? _serverStream;
+
+
+        public RarController(string pipeName)
+        {
+            _pipeName = pipeName;
+            _rarTaskHandler = new RarTaskHandler();
+        }
+
+        public async Task<int> StartAsync(CancellationToken cancellationToken = default)
```

Comment:
this cancellation token should encapsulate a time out to avoid hanging within the while

Reply:
Fixed.